src/test_utils/es/es_test_config.js

```diff
@@ -20,13 +20,40 @@ export const esTestConfig = new class EsTestConfig {
   }
 
   getUrlParts() {
+
+    let testEsUrl;
+    let protocol;
+    let hostname;
+    let port;
+    let username;
+    let password;
+
+    // Allow setting one complete TEST_ES_URL for Es like https://elastic:changeme@myCloudInstance:9200
+    if (process.env.TEST_ES_URL) {
```

Comment:
Can we just return an object here if TEST_ES_URL is defined? Then we can get rid of all the variables and last return.

Reply:
Do you still need/use the other `TEST_ES_*` environment variables? Can we get rid of them if we are now providing the full URL?

Reply:
```js
getUrlParts() {
  // Allow setting one complete TEST_ES_URL for Es like https://elastic:changeme@myCloudInstance:9200
  if (process.env.TEST_ES_URL) {
    const testEsUrl = url.parse(process.env.TEST_ES_URL);
    return {
      protocol: testEsUrl.protocol.slice(0, -1),
      hostname: testEsUrl.hostname,
      port: parseInt(testEsUrl.port, 10 ),
      auth: username + ':' + password,
      username: testEsUrl.username,
      password: testEsUrl.password,
    };
  } else {
    // Allow setting any individual component(s) of the URL,
    // or use default values (username and password from shield.js)
    return {
      protocol: process.env.TEST_ES_PROTOCOL || 'http',
      hostname: process.env.TEST_ES_HOSTNAME || 'localhost',
      port: parseInt(process.env.TEST_ES_PORT, 10) || 9220,
      auth: username + ':' + password,
      username: process.env.TEST_ES_USERNAME || admin.username,
      password: process.env.TEST_ES_PASSWORD || admin.password,
    }
  }
}
```

Reply:
```
getUrlParts() {
  // allow setting one complete TEST_ES_URL for ES like https://elastic:changeme@example.com:9200
  if (process.env.TEST_ES_URL) {
    const testEsUrl = url.parse(process.env.TEST_ES_URL);

    return {
      ...testEsUrl,
      port: parseInt(testEsUrl.port, 10),
      auth: `${testEsUrl.username}:${testEsUrl.password}`,
      testEsUrl.protocol.slice(0, -1),
    };
  }
  
  const username = process.env.TEST_ES_USERNAME || admin.username;
  const password = process.env.TEST_ES_PASSWORD || admin.password;
  return {
    protocol: process.env.TEST_ES_PROTOCOL || 'http',
    hostname: process.env.TEST_ES_HOSTNAME || 'localhost',
    port: parseInt(process.env.TEST_ES_PORT, 10) || 9220,
    auth: `${username}:${password}`,
    username,
    password, 
  };
}
```

Reply:
That sounds pretty good.  I'll try it that way.

Reply:
I'm not sure if I can do that for both the kibana and elasticsearch files;

![image](https://user-images.githubusercontent.com/13542669/38338748-84467036-3830-11e8-80d9-61553469ff70.png)